src/node_buffer.cc

```diff
@@ -565,6 +566,54 @@ void StringSlice(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+// Convert the input into an encoded string
+void DecodeUTF8(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);  // list, flags
+
+  if (!(args[0]->IsArrayBuffer() || args[0]->IsSharedArrayBuffer() ||
+        args[0]->IsArrayBufferView())) {
+    return node::THROW_ERR_INVALID_ARG_TYPE(
+        env->isolate(),
+        "The \"list\" argument must be an instance of SharedArrayBuffer, "
+        "ArrayBuffer or ArrayBufferView.");
+  }
+
+  ArrayBufferViewContents<char> buffer(args[0]);
+
+  CHECK(args[1]->IsBoolean());
+  bool ignore_bom = args[1]->BooleanValue(env->isolate());
+
+  const char* data = buffer.data();
+  auto beginning = 0;
+  auto length = buffer.length();
+
+  if (buffer.length() == 0) return args.GetReturnValue().SetEmptyString();
+
+  if (!ignore_bom) {
+    char bom[] = "\xEF\xBB\xBF";
+
+    if (strncmp(data, bom, 0) == 0 && strncmp(data, bom, 1) == 0 &&
+        strncmp(data, bom, 2) == 0) {
```

Comment:
This should probably be:

```suggestion
    if (strncmp(data, bom, 3) == 0) {
```

since you want to compare three characters, and if you do that, you don't need to compare the first 2 or first 1 characters because comparing all three covers it already

Reply:
Thanks for the review. I updated the commit.